python/tests/spark/test_code_gen.py

```diff
@@ -11,15 +11,60 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 
-from pyspark.sql.session import SparkSession
+from pathlib import Path
 
-from rikai.spark.sql import init
+from pyspark.sql import SparkSession, Row
+from pyspark.sql.types import BinaryType, StructField, StructType
+import torchvision
+import torch
+import numpy as np
 
+from rikai.spark.sql.codegen.fs import FileSystemModel
+from rikai.numpy import wrap
 
-def test_model_codegen_registered(spark: SparkSession):
-    init(spark)
 
+def test_model_codegen_registered(spark: SparkSession):
     spark.sql(
         """CREATE MODEL foo OPTIONS (foo="str",bar=True,max_score=1.23)
          USING 'test://model/a/b/c'"""
     ).count()
+
+
+def test_yaml_spec(spark: SparkSession, tmp_path: Path):
+    spec_yaml = """
+version: 1.0
+name: resnet
+model:
+  uri: resnet.pth
+  flavor: pytorch
+schema: struct<box:box2d, score:float, class:int>
+transforms:
+  pre: demoproject.yolo.transform
+  post: demoproject.yolo.postprocess
```

Comment:
Some thoughts:
* Should we also make the input schema explicit? There's actually several boundaries where one might wish to know the shape/schema of the data: 1) before pre-processing, 2) after pre-processing, 3) before post-processing, 4) after post-processing. If we know 1 and 2 then maybe we can automatically build a pre-processing pipeline (e.g., if it only involves resizing and batching images) but maybe that's wishful thinking (in practice the steps might not be so trivial).
* Should we provide the ability to construct the declaratively in YAML? Something like
  ```
  transforms:
    pre:
      - resize
          width: 600
          height: 600
          mode: fill
      - batch:
          size: 4
  ```
  This way the model creator doesn't have to write imperative preprocessing code.